dataset/src/main/scala/frameless/ops/GroupByOps.scala

```diff
@@ -173,4 +185,52 @@ class GroupedBy2Ops[K1, K2, V](
   def flatMapGroups[U: TypedEncoder](f: ((K1, K2), Iterator[V]) => TraversableOnce[U]): TypedDataset[U] = {
     underlying.flatMapGroups(f)
   }
+
+  def pivot[P: CatalystPivotable, Values <: HList](pivotColumn: TypedColumn[V, P],
+                                values: Values)(
```

Comment:
I think this would be first time we ask users to write an `HList`. Could we avoid it using `SingletonProductArgs`? (We have any example of usage [here](https://github.com/typelevel/frameless/blob/3066e2ecd1f573c67fe112107769fdd8ccf1d34b/dataset/src/main/scala/frameless/TypedDataset.scala#L175-L176))